charts/rstudio-library/files/helm-2/service.tpl

```diff
@@ -20,4 +21,4 @@ spec:
   selector:
     job-name: {{toYaml .Job.id }}
   clusterIP: ''
-  type: NodePort
+  type: {{ $templateData.service.type }}
```

Comment:
Since you are conditional based on the presence of volume mounts, should you be conditional here (and if they don't provide it, then provide a default)?

Reply:
Yeah, we can do that if we want. I'm open to thoughts - I was just counting on the data having the default in it 🤷 We are in control of the defaults in the `values.yaml`, which would be a very natural place to see "what is the default" without having to go read these templates.

Plus it keeps things a smidge simpler in the template
https://github.com/rstudio/helm/blob/2bfb94d8b3331d10a3a78a74040a6b4433a6be4c/charts/rstudio-workbench/values.yaml#L399-L404

Reply:
Totally agree on keeping it simpler. As long as you can control the defaults, I'd lean the direction you had proposed!